src/NuGet.Clients/NuGet.VisualStudio/LegacyTypes/SemanticVersion.cs

```diff
@@ -210,7 +210,7 @@ public int CompareTo(object obj)
             SemanticVersion other = obj as SemanticVersion;
             if (other == null)
             {
-                throw new ArgumentException("obj");
+                throw new ArgumentException("Argument must be of type 'SemanticVersion'.", nameof(obj));
```

Comment:
I feel we need a string resource for this message:

```suggestion
                throw new ArgumentException(string.Format(Strings.ArgumentTypeExceptionMessage, nameof(SemanticVersion)), nameof(obj));
```

Reply:
Added RESX file... do I need to do anything special for new files?

Reply:
NuGet.VisualStudio.dll is already picked up for localization. No further action is required to add the new RESX file. I will keep an eye on it.